system/jlib/jtrace.hpp

```diff
@@ -67,6 +80,9 @@ interface ISpan : extends IInterface
     virtual void toString(StringBuffer & out) const = 0;
     virtual void getLogPrefix(StringBuffer & out) const = 0;
     virtual bool isRecording() const = 0;   // Is it worth adding any events/attributes to this span?
+    virtual void recordException(IException * e, bool spanFailed = true) = 0;
```

Comment:
For consistency I think it would be better if all 3 functions had bool spanFailed as the first parameter.  Requiring the parameter to be provided is not a problem - and probably an advantage.

Reply:
Agreed on consistency where it applies, but these functions are not symmetrical by design.
The leading boolean is required in the setStatus function, and it is optional in the recordError/Exception.

Reply:
ok.  Fine to leave - although in general I am less and less of a fan of optional parameters.